services/web/server/src/simcore_service_webserver/products/_db.py

```diff
@@ -59,6 +59,10 @@ async def get_product(self, product_name: str) -> Product | None:
             )
             row: RowProxy | None = await result.first()
             if row:
+                # NOTE: MD Observation: Currently we are not defensive, we assume automatically
+                # that the product is not billable when there is no product in the products_prices table
+                # or it's price is 0. We should change it and always assume that the product is billable, unless
+                # explicitely stated that it is free
```

Comment:
If this is something that we want maybe create a case as well.

Reply:
I see your point but here is a question: 
Assume you assume the default you propose: all products are billable except if set otherwise. Now, what to do in these situatinos:
- A user wants to do a payment
- We display the price per credit
We fail in all this at runtime?

I propose to achieve what you are saying at startup: All product_prizes MUST be defined for all products, otherwise the service does not start. i.e. a consistency check that is guaranteed at startup.

Reply:
Answers:
- A user wants to make a payment
  - I expect if it is billable then the price should be filled, therefore he can make a payment. Otherwise, something is wrong on our side and yes it should fail -> and we should investigate and fix it. 
-  We display the price per credit
  - Yes we should fail, either at runtime or I agree with you we can catch this at the starting of the application. I think not necessary it needs to be with product_prizes, in my opinion if the product is not billable it does not need to have product_prize, but yes if it is billable it needs to have product_prize. These tables has one-to-many relationship, as product_prizes can change over time.
  - Also I see difference between product is not billable VS product is billable and his product_price is 0. The difference is as follows when product is not billable we do not need to compute credits (which might be the case for osparc.io in the beginning, as for computation of the credits you would need to state the price unit cost for each service which might be demanding. While the product is billable with credit_price 0 means, we compute the credits, we just "give him unlimited credits". 

Sorry if it is confusing, these are just my first thoughts, we can discuss it in person :)